src/Category/Algebra.agda

```diff
@@ -0,0 +1,119 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Algebras for endofunctors
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Category.Category using (Category)
+
+module Category.Algebra {ℓ} (cat : Category ℓ) where
+
+open import Level
+open Category cat
+open import Category.Functor cat cat
+  renaming (RawFunctor to RawEndoFunctor)
+
+module _ {fun : Obj → Obj} (F : RawEndoFunctor fun) where
+  open RawEndoFunctor F
+  
+  record Algebra : Set (suc ℓ) where
+    field
+      Carrier  : Obj
+      evaluate : fun Carrier ⇒ Carrier
+  open Algebra public
+
+  open import Category.Structures
+  module AlgebraCategory (isCat : IsCategory cat) (isFun : IsFunctor cat cat F) where
+    open IsCategory isCat
+    open IsFunctor isFun
+    open ≈-Reasoning
+
+    record AlgebraMorphism (algu : Algebra) (algv : Algebra) : Set ℓ where
+      field
+        translation : Carrier algu ⇒ Carrier algv
+        commutes    : evaluate algv ∘ fmap translation ≈ translation ∘ evaluate algu
+    open AlgebraMorphism
+    
+    idAlgebraMorphism : ∀ (alg : Algebra) → AlgebraMorphism alg alg
+    idAlgebraMorphism alg = record
+      { translation = id
+      ; commutes = begin
+        evaluate alg ∘ fmap id ≈⟨ fmap-identity over evaluate alg ⟩
+        evaluate alg ∘ id      ≈⟨ identityʳ ⟩
+        evaluate alg           ≈⟨ ≈sym identityˡ ⟩
+        id ∘ evaluate alg      ∎
+      }
+
+    infixr 9 _∘alg_
+    _∘alg_ : ∀ {algu algv algw : Algebra}
```

Comment:
This infix notation isn't much use as it lies within an anonymous module with an explicit argument.